src/pages/BooksList.jsx

```diff
@@ -8,35 +8,35 @@ function BooksList() {
   const [loading, setLoading] = useState(true);
   const navigate = useNavigate();
 
-  const getListOfBooks = async () => {
-    try {
-      const res = await axios.post(
-        "https://api-preprod.lelivrescolaire.fr/graph",
-        {
-          query:
-            "query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}",
-        },
-        {
-          headers: {
-            "Content-Type": "application/json; charset=utf-8",
+  useEffect(() => {
+    const getListOfBooks = async () => {
+      try {
+        const res = await axios.post(
+          "https://api-preprod.lelivrescolaire.fr/graph",
+          {
+            query:
+              "query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}",
           },
-        }
-      );
-      return res.data;
-    } catch (err) {
-      console.error("Error:", err.response ? err.response.data : err.message);
+          {
+            headers: {
+              "Content-Type": "application/json; charset=utf-8",
+            },
+          }
+        );
+        return res.data;
+      } catch (err) {
+        console.error("Error:", err.response ? err.response.data : err.message);
 
-      return null;
-    }
-  };
+        return null;
+      }
+    };
 
-  useEffect(() => {
     const fetchData = async () => {
       const booksList = await getListOfBooks();
```

Comment:
This is called every time we get on the page, with no cache. Unfortunately, it's a rather long request.
You could for example use a context to keep the list of books, and not refetch it everytime. It would also possibly avoid passing that much info about the selected book during the navigation.